02_Dynamics/03_Tendon_Robot/Matlab/TendonRobotPDE.m

```diff
@@ -224,27 +224,27 @@
         tau=max(guess(7:7+num_tendons-1),0); %Pull force, which is the largest when compared with 0
         slack=-min(guess(7:7+num_tendons-1),0);
 
-        %pi_b_norm=ones(num_tendons,1)*base_to_motor;
-        pi_b_norm=-Z_t(t)*ones(num_tendons,1);
-        %z=[v;u;q;w;vs_and_us];
-        y0=[p0; reshape(R0,9,1); v0; u0; q0; w0; pi_b_norm];
-       
-        [~,Yout] = ode45(@staticODE, linspace(0,L,200), y0);
-        Y=Yout';
-        for j = 1 : 100-1
-          Z(1:6,j)=Y(13:18,j);
-          Z(13:18,j)=Y(13:18,j);
-        end
+        %pib_s_norm=ones(num_tendons,1)*base_to_motor;
+        pib_s_norm=-Z_t(t)*ones(num_tendons,1);
+        
+        %ODE45
+%         y0=[p0; reshape(R0,9,1); v0; u0; q0; w0; pi_b_norm];
+%         [~,Yout] = ode45(@staticODE, linspace(0,L,200), y0);
+%         Y=Yout';
+%         for j = 1 : N-1
+%           Z(1:6,j)=Y(13:18,j);
+%           Z(13:18,j)=Y(13:18,j);
```

Comment:
`Z(13:18, j) = Y(13:18, j);` is setting the old values of v_s and u_s to be the old values of v and u. So it should be
```
y_s = staticODE(s(j), Y(:, j));
Z(13:18, j) = y_s(13:18);
```
The old v_s and u_s are used to calculate the mixed partials v_st and u_st.

Reply:
You are right, I'm sorry for this error. I'll fix it soon.

Reply:
Nice :)